src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -2048,6 +2050,8 @@ def _get_public_ip_address_allocation(value, sku):
     if secrets:
         secrets = _merge_secrets([validate_file_or_dict(secret) for secret in secrets])
 
+    validate_terminate_notification(terminate_notification, terminate_notification_time)
```

Comment:
This validate function is used only for arguments, right? If yes, it should be executed before command handler and you should specify your argument validator in argument context like https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/vm/_params.py#L238.

Reply:
It seems reasonable.